fedimint-client-wasm/src/lib.rs

```diff
@@ -93,30 +107,60 @@ impl WasmClient {
     #[wasm_bindgen]
     /// Call a fedimint client rpc the responses are returned using `cb`
     /// callback. Each rpc call *can* return multiple responses by calling
-    /// `cb` multiple times. You should ignore the promise by this function
-    /// because it has no significance.
-    pub async fn rpc(&self, module: &str, method: &str, payload: String, cb: &js_sys::Function) {
-        let mut stream = pin!(self.rpc_inner(module, method, payload));
-
-        while let Some(item) = stream.next().await {
-            let this = JsValue::null();
-            let _ = match item {
-                Ok(item) => cb.call1(
-                    &this,
-                    &JsValue::from_str(&serde_json::to_string(&item).unwrap()),
-                ),
-                Err(err) => cb.call1(
-                    &this,
-                    &JsValue::from_str(
-                        &serde_json::to_string(&json!({"error": err.to_string()})).unwrap(),
-                    ),
-                ),
+    /// `cb` multiple times. The returned RpcHandle can be used to cancel the
+    /// operation.
+    pub fn rpc(
+        &self,
+        module: &str,
+        method: &str,
+        payload: String,
+        cb: &js_sys::Function,
+    ) -> RpcHandle {
+        let (abort_handle, abort_registration) = AbortHandle::new_pair();
+        let rpc_handle = RpcHandle { abort_handle };
+
+        let client = self.client.clone();
+        let module = module.to_string();
+        let method = method.to_string();
+        let cb = cb.clone();
+
+        wasm_bindgen_futures::spawn_local(async move {
+            let future = async {
+                let mut stream = pin!(Self::rpc_inner(&client, &module, &method, payload));
+
+                while let Some(item) = stream.next().await {
+                    let this = JsValue::null();
+                    let _ = match item {
+                        Ok(item) => cb.call1(
+                            &this,
+                            &JsValue::from_str(
+                                &serde_json::to_string(&json!({"data": item})).unwrap(),
```

Comment:
Will changing the json key to `"data"` cause issue? We were previously sending `&serde_json::to_string(&item)`

Reply:
yes, this was requested by @alexlwn123